spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/jdbc/AmazonRdsDatabaseAutoConfiguration.java

```diff
@@ -77,10 +77,13 @@ public static RdsInstanceConfigurerBeanPostProcessor rdsInstanceConfigurerBeanPo
 		@Override
 		public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata,
 				BeanDefinitionRegistry registry) {
-			String amazonRdsClientBeanName = AmazonWebserviceClientConfigurationUtils.registerAmazonWebserviceClient(
-					this, registry, "com.amazonaws.services.rds.AmazonRDSClient", null, null).getBeanName();
-
-			rdsDatabaseProperties().getInstances().stream().filter(RdsInstance::hasRequiredPropertiesSet)
+			AmazonRdsDatabaseProperties properties = rdsDatabaseProperties();
+			String amazonRdsClientBeanName = AmazonWebserviceClientConfigurationUtils
+					.registerAmazonWebserviceClient(this, registry, "com.amazonaws.services.rds.AmazonRDSClient", null,
+							properties.getRegion(),
+							properties.getEndpoint() != null ? properties.getEndpoint().toString() : null)
```

Comment:
can we move this to `String endpoint = properties.getEndpoint() != null ? properties.getEndpoint().toString() : null;` and then use endpoint here?